### PR TITLE
Dyn 4977 crash during installation

### PR DESCRIPTION
**Pull request by jesusalvino**
### Purpose

Excluding wrong package versions in order to prevent  Dynamo from Crash
https://jira.autodesk.com/browse/DYN-4977

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

###Reviewers
@QilongTang

###FYIs
@RobertGlobant20 @filipeotero

### CODE REVIEW

File: src/DynamoPackages/PackageManagerClient.cs [hunk 1]
```diff
@@ -77,11 +77,31 @@ internal IEnumerable<PackageHeader> ListAll()
...
 
+        /// <summary>
+        /// For every package Checks its versions and exclude the ones with errors
+        /// </summary>
+        /// <param name="packages"></param>
+        void CleanPackagesWithWrongVersions(List<PackageHeader> packages)
+        {
+            foreach (var package in packages)
+            {
+                bool packageHasWrongVersion = package.versions.Count(v => v.full_dependency_versions == null) > 0;
```
Comment (QilongTang) [inline src/DynamoPackages/PackageManagerClient.cs:94]: @zeusongit Can you remind me what does `full_dependency_versions` mean here?

Comment (zeusongit) [reply src/DynamoPackages/PackageManagerClient.cs:94]: it contains all version of packages listed in `full_dependency_ids` these are packages on which the current package is dependent upon.

Comment (zeusongit) [reply src/DynamoPackages/PackageManagerClient.cs:94]: I talked to greg marr about this. It was probably a DB error where a partial duplication occurred somehow.

Comment (QilongTang) [reply src/DynamoPackages/PackageManagerClient.cs:94]: @zeusongit So this change is expected? Do we need to apply this elsewhere?

Comment (jesusalvino) [reply src/DynamoPackages/PackageManagerClient.cs:94]: There are 70 packages under the same scenario, I have added more details here https://jira.autodesk.com/browse/DYN-4977  , I'm checking in-depth in order to find the root cause

Comment (reddyashish) [reply src/DynamoPackages/PackageManagerClient.cs:94]: Is the packages list here, a list of dependent packages?

Comment (jesusalvino) [reply src/DynamoPackages/PackageManagerClient.cs:94]: > Is the packages list here, a list of dependent packages?

not, they are packages that also have wrong data in their versions.

**Review by reddyashish — APPROVED** [review]

**Comment by QilongTang** [comment]
Regression
```
Dynamo.PackageManager.Tests.PackageManagerClientTests.ListAll_ReturnsNonEmptyListWhenPackagesAreAvailable
```
@zeusongit @reddyashish Regardless of the DPM server side improvement, should we still merge this one on client side?

**Comment by zeusongit** [comment]
I think we should not base our results on `full_dependencies` as described in my previous comment. 

However regarding the failing test it will always fail because the response provided in the test is an empty record, and the new condition will filter it out.
We can add the `url` property in the response so the `ListAll` allows it and then we can have the test pass as usual.

**Comment by QilongTang** [comment]
@zeusongit Make sense, so `url` is more guaranteed to exist than `full_dependencies`?

**Review by zeusongit — APPROVED** [review]
I think you will still need to update the test, other than that LGTM

**Comment by jesusalvino** [comment]
imagen

The related Test has been updated

**Pull request merged**